error on openjdk13 with OpenJ9

# Bug report for FastAsyncWorldEdit 1.15
<!--- If you are using 1.13 or 1.13.1 consider updating to 1.14.4 before raising an issue -->
<!--- The priority lays on 1.14 right now, so issues reported for 1.13 will be fixed for the 1.14 versions -->
<!--- In order to create a valid issue report you have to follow this template. -->
<!--- Remove this template if making a suggestion or asking a question. -->
<!--- Incomplete reports will most likely be marked as invalid, and closed, with few exceptions.-->

**[REQUIRED] FastAsyncWorldEdit Configuration Files:** 
<!--- Issue /fawe debugpaste in game or in your console and copy the supplied URL here -->
<!--- If you cannot perform the above, we require logs/latest.log; config.yml and config-legacy.yml -->
<!--- If you are unwilling to supply the information we need, we reserve the right to not assist you. Redact IP addresses if you need to. -->
default
**[REQUIRED] FastAsyncWorldEdit Version Number:** 
<!--- Enter /fawe version in game or in your console and copy the full output here -->
1.15-80

**[REQUIRED] Spigot/Paper Version Number:** 
<!--- Enter /version ingame or in your console and paste the full output here -->
git-Paper-105

**[REQUIRED] Description of the problem:**
<!--- Check your console for errors while testing -->
<!--- Include relevant information like errors or a picture of the problem -->
<!--- Be as specific as possible.  Don't lie, redact information, or use false names/situations. -->
<!--- Who, What, When, Where, Why, How, Expected behavior, Resultant behavior, etc -->

**How to replicate**:
<!--- If you can reproduce the issue please tell us as detailed as possible step by step how to do that -->
start the server

**Plugins being used on the server:**
<!--- Optional but recommended - issue "/plugins" in-game or in console and copy/paste the list -->

**Checklist**:
<!--- Make sure you've completed the following steps (put an "X" between of brackets): -->
- [] I included all information required in the sections above
- [] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit-1.13/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.14.4](https://ci.athion.net/job/FastAsyncWorldEdit-pipeline/)
- [] I made sure the bug/error is not caused by any other plugin

**log**
```
[08:52:47 INFO]: Got request to register class com.sk89q.worldedit.bukkit.BukkitServerInterface with WorldEdit [com.sk89q.worldedit.extension.platform.PlatformManager@637a4db0]
[08:52:47 INFO]: WEPIF: Using the Bukkit Permissions API.
[08:52:50 INFO]: Using com.sk89q.worldedit.bukkit.adapter.impl.FAWE_Spigot_v1_15_R2 as the Bukkit adapter
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.boydti.fawe.util.ReflectionUtils (file:/root/workdir/plugins/FastAsyncWorldEdit.jar) to field java.lang.Enum.ordinal
WARNING: Please consider reporting this to the maintainers of com.boydti.fawe.util.ReflectionUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[08:53:03 WARN]: java.lang.NoSuchMethodException: java.lang.Class.getDeclaredFields0(boolean)
[08:53:03 WARN]: 	at java.base/java.lang.Class.newNoSuchMethodException(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.throwExceptionOrReturnNull(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.getMethodHelper(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.getDeclaredMethod(Unknown Source)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.setAccessibleNonFinal(ReflectionUtils.java:45)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.setFailsafeFieldValue(ReflectionUtils.java:62)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils9.makeEnum(ReflectionUtils9.java:70)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils9.addEnum(ReflectionUtils9.java:38)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.addEnum(ReflectionUtils.java:22)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.PropertyKey.getOrCreate(PropertyKey.java:107)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.AbstractProperty.<init>(AbstractProperty.java:53)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:38)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:34)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_15_R2.getProperties(Spigot_v1_15_R2.java:486)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.adapter.impl.FAWE_Spigot_v1_15_R2.getProperties(FAWE_Spigot_v1_15_R2.java:330)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.BukkitBlockRegistry.getProperties(BukkitBlockRegistry.java:89)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache$Settings.<init>(BlockTypesCache.java:54)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockType.<init>(BlockType.java:67)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache.register(BlockTypesCache.java:239)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache.<clinit>(BlockTypesCache.java:205)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypes.init(BlockTypes.java:733)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypes.<clinit>(BlockTypes.java:40)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:127)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:105)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.loadFromResource(LegacyMapper.java:112)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.<init>(LegacyMapper.java:76)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.getInstance(LegacyMapper.java:295)
[08:53:03 WARN]: 	at com.sk89q.worldedit.WorldEdit.loadMappings(WorldEdit.java:384)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.setupPreWorldData(WorldEditPlugin.java:240)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.onEnable(WorldEditPlugin.java:219)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.init(WorldEditPlugin.java:167)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.<init>(WorldEditPlugin.java:151)
[08:53:03 WARN]: 	at java.base/java.lang.J9VMInternals.newInstanceImpl(Native Method)
[08:53:03 WARN]: 	at java.base/java.lang.Class.newInstance(Unknown Source)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:80)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:134)
[08:53:03 WARN]: 	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:352)
[08:53:03 WARN]: 	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:272)
[08:53:03 WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.loadPlugins(CraftServer.java:358)
[08:53:03 WARN]: 	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:234)
[08:53:03 WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885)
[08:53:03 WARN]: 	at java.base/java.lang.Thread.run(Unknown Source)
[08:53:03 WARN]: java.lang.RuntimeException: java.lang.Class.getDeclaredFields0(boolean)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils9.addEnum(ReflectionUtils9.java:59)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.addEnum(ReflectionUtils.java:22)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.PropertyKey.getOrCreate(PropertyKey.java:107)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.AbstractProperty.<init>(AbstractProperty.java:53)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:38)
[08:53:03 WARN]: 	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:34)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_15_R2.getProperties(Spigot_v1_15_R2.java:486)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.adapter.impl.FAWE_Spigot_v1_15_R2.getProperties(FAWE_Spigot_v1_15_R2.java:330)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.BukkitBlockRegistry.getProperties(BukkitBlockRegistry.java:89)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache$Settings.<init>(BlockTypesCache.java:54)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockType.<init>(BlockType.java:67)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache.register(BlockTypesCache.java:239)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypesCache.<clinit>(BlockTypesCache.java:205)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypes.init(BlockTypes.java:733)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockTypes.<clinit>(BlockTypes.java:40)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:127)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:105)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.loadFromResource(LegacyMapper.java:112)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.<init>(LegacyMapper.java:76)
[08:53:03 WARN]: 	at com.sk89q.worldedit.world.registry.LegacyMapper.getInstance(LegacyMapper.java:295)
[08:53:03 WARN]: 	at com.sk89q.worldedit.WorldEdit.loadMappings(WorldEdit.java:384)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.setupPreWorldData(WorldEditPlugin.java:240)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.onEnable(WorldEditPlugin.java:219)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.init(WorldEditPlugin.java:167)
[08:53:03 WARN]: 	at com.sk89q.worldedit.bukkit.WorldEditPlugin.<init>(WorldEditPlugin.java:151)
[08:53:03 WARN]: 	at java.base/java.lang.J9VMInternals.newInstanceImpl(Native Method)
[08:53:03 WARN]: 	at java.base/java.lang.Class.newInstance(Unknown Source)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:80)
[08:53:03 WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:134)
[08:53:03 WARN]: 	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:352)
[08:53:03 WARN]: 	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:272)
[08:53:03 WARN]: 	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.loadPlugins(CraftServer.java:358)
[08:53:03 WARN]: 	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:234)
[08:53:03 WARN]: 	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885)
[08:53:03 WARN]: 	at java.base/java.lang.Thread.run(Unknown Source)
[08:53:03 WARN]: Caused by: java.lang.NoSuchMethodException: java.lang.Class.getDeclaredFields0(boolean)
[08:53:03 WARN]: 	at java.base/java.lang.Class.newNoSuchMethodException(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.throwExceptionOrReturnNull(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.getMethodHelper(Unknown Source)
[08:53:03 WARN]: 	at java.base/java.lang.Class.getDeclaredMethod(Unknown Source)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.setAccessibleNonFinal(ReflectionUtils.java:45)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils.setFailsafeFieldValue(ReflectionUtils.java:62)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils9.makeEnum(ReflectionUtils9.java:70)
[08:53:03 WARN]: 	at com.boydti.fawe.util.ReflectionUtils9.addEnum(ReflectionUtils9.java:38)
[08:53:03 WARN]: 	... 35 more
[08:53:03 WARN]: Failed to load the built-in legacy id registry
java.lang.ExceptionInInitializerError: null
	at java.lang.J9VMInternals.ensureError(Unknown Source) ~[?:?]
	at java.lang.J9VMInternals.recordInitializationFailure(Unknown Source) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypes.init(BlockTypes.java:733) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypes.<clinit>(BlockTypes.java:40) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:127) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockState.get(BlockState.java:105) ~[?:?]
	at com.sk89q.worldedit.world.registry.LegacyMapper.loadFromResource(LegacyMapper.java:112) ~[?:?]
	at com.sk89q.worldedit.world.registry.LegacyMapper.<init>(LegacyMapper.java:76) ~[?:?]
	at com.sk89q.worldedit.world.registry.LegacyMapper.getInstance(LegacyMapper.java:295) ~[?:?]
	at com.sk89q.worldedit.WorldEdit.loadMappings(WorldEdit.java:384) ~[?:?]
	at com.sk89q.worldedit.bukkit.WorldEditPlugin.setupPreWorldData(WorldEditPlugin.java:240) ~[?:?]
	at com.sk89q.worldedit.bukkit.WorldEditPlugin.onEnable(WorldEditPlugin.java:219) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[papermc.jar:git-Paper-105]
	at com.sk89q.worldedit.bukkit.WorldEditPlugin.init(WorldEditPlugin.java:167) ~[?:?]
	at com.sk89q.worldedit.bukkit.WorldEditPlugin.<init>(WorldEditPlugin.java:151) ~[?:?]
	at java.lang.J9VMInternals.newInstanceImpl(Native Method) ~[?:?]
	at java.lang.Class.newInstance(Unknown Source) ~[?:?]
	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:80) ~[papermc.jar:git-Paper-105]
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:134) ~[papermc.jar:git-Paper-105]
	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:352) ~[papermc.jar:git-Paper-105]
	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:272) ~[papermc.jar:git-Paper-105]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.loadPlugins(CraftServer.java:358) ~[papermc.jar:git-Paper-105]
	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:234) ~[papermc.jar:git-Paper-105]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885) ~[papermc.jar:git-Paper-105]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.Class.getDeclaredFields0(boolean)
	at com.sk89q.worldedit.world.block.BlockTypesCache.<clinit>(BlockTypesCache.java:230) ~[?:?]
	... 23 more
Caused by: java.lang.RuntimeException: java.lang.Class.getDeclaredFields0(boolean)
	at com.boydti.fawe.util.ReflectionUtils9.addEnum(ReflectionUtils9.java:59) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils.addEnum(ReflectionUtils.java:22) ~[?:?]
	at com.sk89q.worldedit.registry.state.PropertyKey.getOrCreate(PropertyKey.java:107) ~[?:?]
	at com.sk89q.worldedit.registry.state.AbstractProperty.<init>(AbstractProperty.java:53) ~[?:?]
	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:38) ~[?:?]
	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:34) ~[?:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_15_R2.getProperties(Spigot_v1_15_R2.java:486) ~[?:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.FAWE_Spigot_v1_15_R2.getProperties(FAWE_Spigot_v1_15_R2.java:330) ~[?:?]
	at com.sk89q.worldedit.bukkit.BukkitBlockRegistry.getProperties(BukkitBlockRegistry.java:89) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache$Settings.<init>(BlockTypesCache.java:54) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockType.<init>(BlockType.java:67) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache.register(BlockTypesCache.java:239) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache.<clinit>(BlockTypesCache.java:205) ~[?:?]
	... 23 more
Caused by: java.lang.NoSuchMethodException: java.lang.Class.getDeclaredFields0(boolean)
	at java.lang.Class.newNoSuchMethodException(Unknown Source) ~[?:?]
	at java.lang.Class.throwExceptionOrReturnNull(Unknown Source) ~[?:?]
	at java.lang.Class.getMethodHelper(Unknown Source) ~[?:?]
	at java.lang.Class.getDeclaredMethod(Unknown Source) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils.setAccessibleNonFinal(ReflectionUtils.java:45) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils.setFailsafeFieldValue(ReflectionUtils.java:62) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils9.makeEnum(ReflectionUtils9.java:70) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils9.addEnum(ReflectionUtils9.java:38) ~[?:?]
	at com.boydti.fawe.util.ReflectionUtils.addEnum(ReflectionUtils.java:22) ~[?:?]
	at com.sk89q.worldedit.registry.state.PropertyKey.getOrCreate(PropertyKey.java:107) ~[?:?]
	at com.sk89q.worldedit.registry.state.AbstractProperty.<init>(AbstractProperty.java:53) ~[?:?]
	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:38) ~[?:?]
	at com.sk89q.worldedit.registry.state.IntegerProperty.<init>(IntegerProperty.java:34) ~[?:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.Spigot_v1_15_R2.getProperties(Spigot_v1_15_R2.java:486) ~[?:?]
	at com.sk89q.worldedit.bukkit.adapter.impl.FAWE_Spigot_v1_15_R2.getProperties(FAWE_Spigot_v1_15_R2.java:330) ~[?:?]
	at com.sk89q.worldedit.bukkit.BukkitBlockRegistry.getProperties(BukkitBlockRegistry.java:89) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache$Settings.<init>(BlockTypesCache.java:54) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockType.<init>(BlockType.java:67) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache.register(BlockTypesCache.java:239) ~[?:?]
	at com.sk89q.worldedit.world.block.BlockTypesCache.<clinit>(BlockTypesCache.java:205) ~[?:?]
	... 23 more
```

--- a/worldedit-core/src/main/java/com/boydti/fawe/util/ReflectionUtils.java
+++ b/worldedit-core/src/main/java/com/boydti/fawe/util/ReflectionUtils.java
@@ -1,5 +1,6 @@
 package com.boydti.fawe.util;
 
+import java.lang.invoke.MethodHandles;
 import java.lang.reflect.AccessibleObject;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
@@ -32,26 +33,15 @@ public class ReflectionUtils {
         // letting us modify the static final field
         if (Modifier.isFinal(field.getModifiers())) {
             try {
-                Field modifiersField = Field.class.getDeclaredField("modifiers");
-                modifiersField.setAccessible(true);
-                int modifiers = modifiersField.getInt(field);
+                Field lookupField = MethodHandles.Lookup.class.getDeclaredField("IMPL_LOOKUP");
+                lookupField.setAccessible(true);
 
                 // blank out the final bit in the modifiers int
-                modifiers &= ~Modifier.FINAL;
-                modifiersField.setInt(field, modifiers);
-            } catch (NoSuchFieldException e) {
-                // Java 12+ compatibility - search fields with hidden method for modifiers
-                // same concept as above, just with a more hacky way of getting to the modifiers
-                Method getDeclaredFields0 = Class.class.getDeclaredMethod("getDeclaredFields0", boolean.class);
-                getDeclaredFields0.setAccessible(true);
-                Field[] fields = (Field[]) getDeclaredFields0.invoke(Field.class, false);
-                for (Field classField : fields) {
-                    if ("modifiers".equals(classField.getName())) {
-                        classField.setAccessible(true);
-                        classField.set(field, field.getModifiers() & ~Modifier.FINAL);
-                        break;
-                    }
-                }
+                ((MethodHandles.Lookup) lookupField.get(null))
+                .findSetter(Field.class, "modifiers", int.class)
+                    .invokeExact(field, field.getModifiers() & ~Modifier.FINAL);
+            } catch (Throwable e) {
+                e.printStackTrace();
             }
         }
     }

